syntax = "proto3";

package post;

service PostService {
  rpc CreatePost(CreatePostRequest) returns (Post);
  rpc GetPost(GetPostRequest) returns (Post);
  rpc ListPosts(ListPostsRequest) returns (ListPostsResponse);
  rpc UpdatePost(UpdatePostRequest) returns (Post);
  rpc DeletePost(DeletePostRequest) returns (DeletePostResponse);
}

message CreatePostRequest {
  string user_id = 1;
  string title = 2;
  string content = 3;
}

message GetPostRequest { string post_id = 1; }

message ListPostsRequest {
  string user_id = 1;
  int32 page = 2;
  int32 page_size = 3;
}

message ListPostsResponse {
  repeated Post posts = 1;
  int32 total = 2;
}

message UpdatePostRequest {
  string post_id = 1;
  string title = 2;
  string content = 3;
}

message DeletePostRequest {
  string post_id = 1;
  string user_id = 2; // For authorization
}

message DeletePostResponse { bool success = 1; }

message Post {
  string id = 1;
  string user_id = 2;
  string title = 3;
  string content = 4;
  string created_at = 5;
  string updated_at = 6;
}
